ci/deploy/linux.py

```diff
@@ -58,6 +61,32 @@ def publish_deb(package: str, version: str) -> None:
     )
     bt.repo("apt").package(package).publish_uploads(version)
 
+    s3 = boto3.client("s3")
+    bucket = "materialize-apt-repository"
+    object_key = f"pool/stable/m/ma/materialized-{version}.deb"
+    # Download the staged package (deb-s3 needs to get various metadata from it)
+    s3.download_file(bucket, object_key, f"materialized-{version}.deb")
+    # Import the private key into GPG
+    key = os.environ["GPG_KEY"].replace("\\n", "\n")
+    gpg = subprocess.Popen(["gpg", "--import"], stdin=subprocess.PIPE)
+    gpg.communicate(key.encode("ascii"))
```

Comment:
`Popen` is roughly deprecated in favor of `subprocess.run`, though in this case I think you can just use `spawn.runv`?

```python
spawn.runv(["gpg", "--import"], stdin=io.BytesIO(os.environ["GPG_KEY"]))
```

Reply:
No, you can't use BytesIO/StringIO as input to subprocess.run .

```
>>> subprocess.run(["cat"], stdin=io.BytesIO(b"test"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 505, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 829, in __init__
    errread, errwrite) = self._get_handles(stdin, stdout, stderr)
  File "/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 1586, in _get_handles
    p2cread = stdin.fileno()
io.UnsupportedOperation: fileno
```

Reply:
Ugh. Ok. Sorry to mislead you.

Reply:
It's fine, that was the first thing I tried also